File: emisor-js/capas/presentacion.js
/**
 * Codifica cada carácter a su ASCII binario de 8 bits.
 * @param {string} texto
 * @returns {string} string de bits, ej. "A" -> "01000001"
 */
export function codificarMensaje(texto) {
  return [...texto].map((c) => c.charCodeAt(0).toString(2).padStart(8, "0")).join("");
}

/**
 * Decodifica un string de bits (múltiplo de 8) a texto.
 * @param {string} binario - bits del mensaje, sin bits de redundancia
 * @param {boolean} hayError - si true, no se pudo garantizar integridad
 * @returns {string | null} texto decodificado, o null si hayError es true
 */
export function decodificarMensaje(binario, hayError) {
  if (hayError) return null;
  const bytes = binario.match(/.{1,8}/g) || [];
  return bytes.map((b) => String.fromCharCode(parseInt(b, 2))).join("");
}

File: receptor-py/capas/presentacion.py
def codificar_mensaje(texto):
    """
    Codifica cada carácter de texto a su ASCII binario de 8 bits.

    Args:
        texto (str): mensaje en texto plano.
    Returns:
        str: string de bits, ej. "A" -> "01000001"
    """
    return "".join(format(ord(c), "08b") for c in texto)


def decodificar_mensaje(binario, hay_error):
    """
    Decodifica un string de bits (múltiplos de 8) a texto.

    Args:
        binario (str): bits del mensaje, sin bits de redundancia.
        hay_error (bool): si True, no se pudo garantizar integridad.
    Returns:
        str | None: texto decodificado, o None si hay_error es True
                     (capas/aplicacion.py debe mostrar el error en ese caso).
    """
    if hay_error:
        return None
    caracteres = [binario[i:i + 8] for i in range(0, len(binario), 8)]
    return "".join(chr(int(b, 2)) for b in caracteres)
